merged-packages/json-rpc-engine/tsconfig.json

```
@@ -1,16 +1,8 @@
 {
+  "extends": "../../tsconfig.packages.json",
   "compilerOptions": {
-    "esModuleInterop": true,
-    "exactOptionalPropertyTypes": true,
-    "forceConsistentCasingInFileNames": true,
-    "lib": ["ES2020"],
-    "module": "CommonJS",
-    "moduleResolution": "node",
-    "noEmit": true,
-    "noErrorTruncation": true,
-    "noUncheckedIndexedAccess": true,
-    "strict": true,
-    "target": "es2020"
+    "baseUrl": "./",
+    "rootDir": "../..",
```

As per the discussion in https://github.com/MetaMask/eth-json-rpc-provider/pull/27, we're preserving the TypeScript compiler flags enabled in the original repo instead of downgrading type safety by only using the flags enabled in core.

Of course, any flags or options already enabled in root don't need to be duplicated in subpackage configs.

You can check https://github.com/MetaMask/core/issues/1551#issuecomment-1745665740 for updated info on the migration process. 

```suggestion
    "baseUrl": "./",
    "rootDir": "../..",
    "exactOptionalPropertyTypes": true,
    "forceConsistentCasingInFileNames": true,
    "noErrorTruncation": true,
    "noUncheckedIndexedAccess": true,
    "target": "es2020"
```

@mcmire Just to confirm, should we keep TypeScript compilation target versions that are higher than root? For `eth-json-rpc-provider`, I kept the es2017 (es8) target instead of extending the es6 setting in root.

---

No I don't see any issues with doing this. It should be fine to do so.